src/test/java/com/pubnub/contract/subscribe/steps/ThenSteps.kt

```diff
@@ -0,0 +1,51 @@
+package com.pubnub.contract.subscribe.steps
+
+import com.pubnub.api.models.consumer.pubsub.PNMessageResult
+import com.pubnub.api.models.consumer.pubsub.PNSignalResult
+import com.pubnub.contract.subscribe.state.SubscribeState
+import io.cucumber.java.en.Then
+import org.awaitility.Awaitility.await
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.*
+import org.junit.jupiter.api.Assertions.assertTrue
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+class ThenSteps(
+    private val subscribeState: SubscribeState
+) {
+
+    @Then("I receive 2 messages in my subscribe response")
+    fun I_receive_2_messages_in_my_subscribe_response() {
+        await()
+            .atMost(500, TimeUnit.MILLISECONDS)
+            .until {
+                subscribeState.messages.size == 2
+            }
+    }
+
+    @Then("response contains messages with {string} and {string} message types")
+    fun response_contains_messages_with_message_types(
+        messageTypeOfFirstMessage: String,
+        messageTypeOfSecondMessage: String
+    ) {
+        assertThat(subscribeState.messages.size, equalTo(2))
+        val listOfReceivedMessageTypes = subscribeState.messages.map { it.messageType.value }
+        assertThat(
+            listOfReceivedMessageTypes,
+            containsInAnyOrder(messageTypeOfFirstMessage, messageTypeOfSecondMessage)
+        )
+    }
+
+    @Then("response contains messages without space ids")
+    fun response_contains_messages_without_space_ids() {
+        assertTrue(
```

Comment:
Whenever I can I try to avoid true/false assertion. Especially when dealing with collections. Here's the assertion that should check similar things but in case of fail it will also list the spaceIds that are not null giving us more context.

```
 assertThat(
            subscribeState.messagesList.mapNotNull {
                when (it) {
                    is MessageResult -> {
                        it.spaceId
                    }

                    else -> null
                }
            },
            empty()
        )
```

Reply:
For me this is too complicated